ESP32/main/src/wifi/WifiController.cpp

```diff
@@ -0,0 +1,213 @@
+}
+
+void setWifiConfig(String SSID,String PWD,bool ap)
+{
+    ESP_LOGI(TAG,"mssid:%s pw:%s ap:%s", mSSID,mPWD,boolToChar(ap));
+    mSSID = SSID;
+    mPWD = PWD;
+    mAP = ap;
+}
+
+void createAp(String ssid, String password)
+{
+    WiFi.disconnect();
+    ESP_LOGI(TAG,"Ssid %s pw %s",ssid,password);
+    
+    if (ssid.isEmpty())
+    {
+      ESP_LOGI(TAG,"Ssid empty, start Uc2 open softap");
+      WiFi.softAP(mSSIDAP.c_str());
+    }
+    else if (password.isEmpty())
+    {
+      ESP_LOGI(TAG,"Ssid start %s open softap",ssid);
+      WiFi.softAP(ssid.c_str());
+    }
+    else
+    {
+      ESP_LOGI(TAG,"Ssid start %s close softap",ssid);
+      WiFi.softAP(ssid.c_str(), password.c_str());
+    }
+    ESP_LOGI(TAG,"Connected. IP: %s",WiFi.softAPIP().toString());
+    //server = new WebServer(WiFi.softAPIP(),80);
+}
+
+void setup()
+{
+  if (mSSID != nullptr)
+    ESP_LOGI(TAG,"mssid:%s pw:%s ap:%s", mSSID,mPWD,boolToChar(mAP));
+  if (server !=  nullptr)
+  {
+     server->close();
+     server = nullptr;
+  }
+  if(mSSID == "")
+  {
+    mAP = true;
+    createAp(mSSIDAP,mPWD);
+  }
+  else if(mAP){
+    createAp(mSSID,mPWD);
+  }
+  else
+  {
+    WiFi.softAPdisconnect();
+    ESP_LOGI(TAG,"Connect to:%s",mSSID);
+    WiFi.begin(mSSID.c_str(), mPWD.c_str());
+
+    int nConnectTrials = 0;
+    while (WiFi.status() != WL_CONNECTED && nConnectTrials <=10)
+    {
+      ESP_LOGI(TAG,"Wait for connection");
+      delay(200);
+      nConnectTrials += 1;
+      // we can even make the ESP32 to sleep
+    }
+    if (nConnectTrials == 10)
+    {
+      ESP_LOGI(TAG,"failed to connect,Start softap");
+      mAP = true;
+      mSSID = mSSIDAP;
+      mPWD = "";
+      createAp(mSSIDAP,"");
+    }
+    else
+    {
+      ESP_LOGI(TAG,"Connected. IP: %s",WiFi.localIP());
+      //server = new WebServer(WiFi.localIP(),80);
+    }
+  }
+  server = new WebServer(80);
+  if (jsonDocument == nullptr)
+  {
+    createJsonDoc();
+  }
+  setup_routing();
+  server->begin();
+  ESP_LOGI(TAG,"HTTP Running server  nullptr: %s jsondoc  nullptr: %s", boolToChar(server == nullptr),boolToChar(jsonDocument == nullptr));
+}
+
+void setup_routing()
+{
+  ESP_LOGI(TAG,"Setting up HTTP Routing");
+  server->onNotFound(RestApi::handleNotFound);
+  server->on(state_act_endpoint, HTTP_POST, RestApi::State_act);
+  server->on(state_get_endpoint, HTTP_POST, RestApi::State_get);
+  server->on(state_set_endpoint, HTTP_POST, RestApi::State_set);
+
+  server->on(identity_endpoint, RestApi::getIdentity);
+
+  server->on(ota_endpoint, HTTP_GET, RestApi::ota);
+  /*handling uploading firmware file */
+  server->on(update_endpoint, HTTP_POST, RestApi::update, RestApi::upload);
+
+  server->on("/",HTTP_GET, RestApi::getEndpoints);
+  server->on(scanwifi_endpoint, HTTP_GET, RestApi::scanWifi);
+  server->on(connectwifi_endpoint, HTTP_POST, RestApi::connectToWifi);
+  server->on(reset_nv_flash_endpoint, HTTP_GET, RestApi::resetNvFLash);
+
+  // POST
+#ifdef IS_MOTOR
```

Comment:
For serving the entry points, there is no way to have that inside the different "hardware drivers", or? Something like "Laser implements .. blabla" Would be cool to have everything in one `cpp`-file. But this (how you wrote it) is great for now!